Can't run mvn checkstyle:checkstyle

The default checkstyle plugin fails to start when running `mvn checkstyle:checkstyle`:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.14:checkstyle (default-cli) on project moodcat-core: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: Unable to find suppressions file at location: configuration/checkstyle-suppressions.xml: Could not find resource 'configuration/checkstyle-suppressions.xml'
```

I can't find it in the report as well.
Fix in #93?

--- a/pom.xml
+++ b/pom.xml
@@ -14,21 +14,22 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
+		<checkstyle.version>2.15</checkstyle.version>
+		<cobertura.version>2.7</cobertura.version>
+		<findbugs.version>3.0.1</findbugs.version>
 		<guice.version>3.0</guice.version>
 		<hibernate.version>4.3.8.Final</hibernate.version>
 		<jackson.version>2.4.4</jackson.version>
 		<jetty.version>9.2.10.v20150310</jetty.version>
-		<resteasy.version>3.0.11.Final</resteasy.version>
-		<slf4j.version>1.7.7</slf4j.version>
-		<logback.version>1.0.13</logback.version>
 		<junit.version>4.12</junit.version>
-		<mockito.version>1.10.19</mockito.version>
+		<logback.version>1.0.13</logback.version>
 		<lombok.version>1.16.4</lombok.version>
-		<checkstyle.version>2.14</checkstyle.version>
-		<pmd.version>3.2</pmd.version>
-		<findbugs.version>3.0.1</findbugs.version>
+		<mockito.version>1.10.19</mockito.version>
+		<pmd.version>3.4</pmd.version>
 		<querydsl.version>3.6.0</querydsl.version>
 		<recommendation-library.version>2.0.0</recommendation-library.version>
+		<resteasy.version>3.0.11.Final</resteasy.version>
+		<slf4j.version>1.7.7</slf4j.version>
 	</properties>
 
 	<repositories>
@@ -197,6 +198,14 @@
 			<artifactId>resteasy-client</artifactId>
 			<version>${resteasy.version}</version>
 		</dependency>
+
+		<!-- This dependency is needed in order to let JavaDoc not throw an exception 
+			on the @Transactional annotation. -->
+		<dependency>
+			<groupId>javax.interceptor</groupId>
+			<artifactId>javax.interceptor-api</artifactId>
+			<version>1.2</version>
+		</dependency>
 	</dependencies>
 
 	<build>
@@ -220,24 +229,6 @@
 				</configuration>
 			</plugin>
 			<plugin>
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-checkstyle-plugin</artifactId>
-				<version>${checkstyle.version}</version>
-				<configuration>
-					<failOnViolation>false</failOnViolation>
-					<configLocation>configuration/checkstyle.xml</configLocation>
-					<suppressionsLocation>configuration/checkstyle-suppressions.xml</suppressionsLocation>
-				</configuration>
-			</plugin>
-			<plugin>
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-pmd-plugin</artifactId>
-				<version>${pmd.version}</version>
-				<configuration>
-					<format>xml</format>
-				</configuration>
-			</plugin>
-			<plugin>
 				<groupId>com.mysema.maven</groupId>
 				<artifactId>maven-apt-plugin</artifactId>
 				<version>1.0.4</version>
@@ -258,16 +249,49 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-checkstyle-plugin</artifactId>
 				<version>${checkstyle.version}</version>
+				<dependencies>
+					<dependency>
+						<groupId>com.puppycrawl.tools</groupId>
+						<artifactId>checkstyle</artifactId>
+						<version>6.6</version>
+					</dependency>
+				</dependencies>
 				<configuration>
+					<propertyExpansion>basedir=${basedir}</propertyExpansion>
 					<failOnViolation>false</failOnViolation>
-					<configLocation>configuration/checkstyle.xml</configLocation>
-					<suppressionsLocation>configuration/checkstyle-suppressions.xml</suppressionsLocation>
+					<configLocation>src/main/resources/configuration/checkstyle.xml</configLocation>
+					<suppressionsLocation>src/main/resources/configuration/checkstyle-suppressions.xml</suppressionsLocation>
 				</configuration>
+			</plugin>
+			
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>cobertura-maven-plugin</artifactId>
+				<version>${cobertura.version}</version>
+				<dependencies>
+					<dependency>
+						<groupId>org.ow2.asm</groupId>
+						<artifactId>asm</artifactId>
+						<version>5.0.3</version>
+					</dependency>
+				</dependencies>
 			</plugin>
 		</plugins>
 	</build>
 	<reporting>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-checkstyle-plugin</artifactId>
+				<version>${checkstyle.version}</version>
+				<configuration>
+					<propertyExpansion>basedir=${basedir}</propertyExpansion>
+					<failOnViolation>false</failOnViolation>
+					<configLocation>src/main/resources/configuration/checkstyle.xml</configLocation>
+					<suppressionsLocation>src/main/resources/configuration/checkstyle-suppressions.xml</suppressionsLocation>
+				</configuration>
+			</plugin>
+
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-project-info-reports-plugin</artifactId>
@@ -277,7 +301,7 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-javadoc-plugin</artifactId>
-				<version>2.10.1</version>
+				<version>2.10.3</version>
 				<configuration>
 					<failOnError>false</failOnError>
 				</configuration>
@@ -301,15 +325,18 @@
 			<plugin>
 				<groupId>org.codehaus.mojo</groupId>
 				<artifactId>cobertura-maven-plugin</artifactId>
-				<version>2.7</version>
+				<version>${cobertura.version}</version>
 			</plugin>
 
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-pmd-plugin</artifactId>
-				<version>3.4</version>
+				<version>${pmd.version}</version>
 				<configuration>
 					<skipEmptyReport>false</skipEmptyReport>
+					<rulesets>
+						<ruleset>src/main/resources/configuration/pmd-rules.xml</ruleset>
+					</rulesets>
 				</configuration>
 			</plugin>
 

--- a/pom.xml
+++ b/pom.xml
@@ -1,12 +1,58 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 	<modelVersion>4.0.0</modelVersion>
+
 	<groupId>me.moodcat</groupId>
 	<artifactId>moodcat-core</artifactId>
-	<packaging>jar</packaging>
 	<version>1.0-SNAPSHOT</version>
+	<packaging>jar</packaging>
+
 	<name>MoodCat Core</name>
 	<url>http://moodcat.me</url>
+	<description>MoodCat allows users (and friends!) to unite in rooms, and listen to and discuss music together.</description>
+
+	<organization>
+		<name>TU Delft</name>
+		<url>http://ewi.tudelft.nl</url>
+	</organization>
+
+	<scm>
+		<connection>scm:git:git@github.com:MoodCat/MoodCat.me-Core.git</connection>
+		<developerConnection>scm:git:git@github.com:MoodCat/MoodCat.me-Core.git</developerConnection>
+		<url>https://github.com/MoodCat/MoodCat.me-Core</url>
+		<tag>HEAD</tag>
+	</scm>
+
+	<licenses>
+		<license>
+			<name>MIT License</name>
+			<url>http://www.opensource.org/licenses/mit-license.php</url>
+			<distribution>repo</distribution>
+		</license>
+	</licenses>
+
+	<developers>
+		<developer>
+			<name>Eva Anker</name>
+			<email>E.S.Anker@student.tudelft.nl</email>
+		</developer>
+		<developer>
+			<name>Jan-Willem Gmelig Meyling</name>
+			<email>j.gmeligmeyling@student.tudelft.nl</email>
+		</developer>
+		<developer>
+			<name>Jaap Heijligers</name>
+			<email>J.A.Heijligers@student.tudelft.nl</email>
+		</developer>
+		<developer>
+			<name>Tim van der Lippe</name>
+			<email>T.J.vanderLippe@student.tudelft.nl</email>
+		</developer>
+		<developer>
+			<name>Gijs Weterings</name>
+			<email>G.Weterings@student.tudelft.nl</email>
+		</developer>
+	</developers>
 
 	<properties>
 		<maven.compiler.source>1.8</maven.compiler.source>
@@ -228,6 +274,7 @@
 					<target>1.8</target>
 				</configuration>
 			</plugin>
+
 			<plugin>
 				<groupId>com.mysema.maven</groupId>
 				<artifactId>maven-apt-plugin</artifactId>
@@ -261,6 +308,29 @@
 					<failOnViolation>false</failOnViolation>
 					<configLocation>src/main/resources/configuration/checkstyle.xml</configLocation>
 					<suppressionsLocation>src/main/resources/configuration/checkstyle-suppressions.xml</suppressionsLocation>
+					<sourceDirectory>target/generated-sources/delombok</sourceDirectory>
+				</configuration>
+			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-pmd-plugin</artifactId>
+				<version>${pmd.version}</version>
+				<configuration>
+					<skipEmptyReport>false</skipEmptyReport>
+					<rulesets>
+						<ruleset>src/main/resources/configuration/pmd-rules.xml</ruleset>
+					</rulesets>
+					<includeTests>true</includeTests>
+				</configuration>
+			</plugin>
+
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>findbugs-maven-plugin</artifactId>
+				<version>${findbugs.version}</version>
+				<configuration>
+					<failOnError>false</failOnError>
 				</configuration>
 			</plugin>
 			
@@ -276,10 +346,56 @@
 					</dependency>
 				</dependencies>
 			</plugin>
+
+			<!-- The Lombok plugin compiles lombok annotations to the target/generated-sources folder
+				so that they can be used for static analysis -->
+			<plugin>
+				<groupId>org.projectlombok</groupId>
+				<artifactId>lombok-maven-plugin</artifactId>
+				<version>1.16.4.1</version>
+				<executions>
+					<execution>
+						<id>delombok</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>delombok</goal>
+						</goals>
+						<configuration>
+							<addOutputDirectory>false</addOutputDirectory>
+							<sourceDirectory>src/main/java</sourceDirectory>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+
 		</plugins>
 	</build>
+
 	<reporting>
 		<plugins>
+			<!-- Link sources for reports -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jxr-plugin</artifactId>
+				<configuration>
+					<linkJavadoc>true</linkJavadoc>
+				</configuration>
+				<version>2.5</version>
+			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-project-info-reports-plugin</artifactId>
+				<version>2.8</version>
+			</plugin>
+
+			<!-- JUnit report -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-surefire-report-plugin</artifactId>
+				<version>2.18.1</version>
+			</plugin>
+
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-checkstyle-plugin</artifactId>
@@ -294,16 +410,11 @@
 
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-project-info-reports-plugin</artifactId>
-				<version>2.8</version>
-			</plugin>
-
-			<plugin>
-				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-javadoc-plugin</artifactId>
 				<version>2.10.3</version>
 				<configuration>
-					<failOnError>false</failOnError>
+					<failOnError>true</failOnError>
+					<sourcepath>target/generated-sources/delombok</sourcepath>
 				</configuration>
 				<reportSets>
 					<reportSet>
@@ -313,13 +424,6 @@
 						</reports>
 					</reportSet>
 				</reportSets>
-			</plugin>
-
-
-			<plugin> <!-- JUnit report -->
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-surefire-report-plugin</artifactId>
-				<version>2.18.1</version>
 			</plugin>
 
 			<plugin>
@@ -337,6 +441,7 @@
 					<rulesets>
 						<ruleset>src/main/resources/configuration/pmd-rules.xml</ruleset>
 					</rulesets>
+					<includeTests>true</includeTests>
 				</configuration>
 			</plugin>
 
@@ -348,6 +453,39 @@
 					<failOnError>false</failOnError>
 				</configuration>
 			</plugin>
+
 		</plugins>
 	</reporting>
+
+	<!-- Profile is required for the lombok maven plugin.
+			see: http://awhitford.github.io/lombok.maven/lombok-maven-plugin/usage.html -->
+	<profiles>
+		<profile>
+			<id>lombok-needs-tools-jar</id>
+			<activation>
+				<file>
+					<exists>${java.home}/../lib/tools.jar</exists>
+				</file>
+			</activation>
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.projectlombok</groupId>
+						<artifactId>lombok-maven-plugin</artifactId>
+						<version>1.16.4.1</version>
+						<dependencies>
+							<dependency>
+								<groupId>sun.jdk</groupId>
+								<artifactId>tools</artifactId>
+								<version>1.6</version>
+								<scope>system</scope>
+								<systemPath>${java.home}/../lib/tools.jar</systemPath>
+							</dependency>
+						</dependencies>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
+	</profiles>
+
 </project>

--- a/src/main/java/me/moodcat/api/SongAPI.java
+++ b/src/main/java/me/moodcat/api/SongAPI.java
@@ -118,9 +118,9 @@ public class SongAPI {
             @PathParam("vote") final String vote) throws InvalidVoteException {
         final Song song = this.songDAO.findBySoundCloudId(soundCloudId);
 
-        if (vote.equals("like")) {
+        if ("like".equals(vote)) {
             song.increaseNumberOfPositiveVotes();
-        } else if (vote.equals("dislike")) {
+        } else if ("dislike".equals(vote)) {
             song.decreaseNumberOfPositiveVotes();
         } else {
             throw new InvalidVoteException();
@@ -146,8 +146,7 @@ public class SongAPI {
         final VAVector scaledDistance = classificationVector.subtract(songVector)
                 .multiply(CLASSIFICATION_WEIGHT);
 
-        final VAVector updatedVector = songVector.add(scaledDistance);
-        return updatedVector;
+        return songVector.add(scaledDistance);
     }
 
     /**

--- a/src/main/java/me/moodcat/core/App.java
+++ b/src/main/java/me/moodcat/core/App.java
@@ -1,6 +1,8 @@
 package me.moodcat.core;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
 import java.util.EnumSet;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
@@ -69,9 +71,18 @@ public class App {
 
     /**
      * Instantiates the server and adds handlers for the requests.
+     *
+     * @throws IOException
+     *             If the statics folder threw an IOException.
      */
-    public App() {
-        final File staticsFolder = new File("src/main/resources/static/app");
+    public App() throws IOException {
+        final File staticsFolder = new File(Paths.get(".").toAbsolutePath().toString()
+                + "/src/main/resources/static/app");
+
+        // Make sure the folder is available, else we can't start the server.
+        if (!staticsFolder.exists() && !staticsFolder.mkdir()) {
+            throw new IOException("Static folder could not be initialized.");
+        }
 
         for (final String file : staticsFolder.list()) {
             log.info("Found resource {}", file);
@@ -90,7 +101,7 @@ public class App {
      * @throws Exception
      *             Thrown when a thread-exception occured.
      */
-    public static void main(final String[] args) throws Exception {
+    public static void main(final String... args) throws Exception {
         SLF4JBridgeHandler.removeHandlersForRootLogger();
         SLF4JBridgeHandler.install();
 
@@ -191,7 +202,7 @@ public class App {
              */
             private final Module[] overrides;
 
-            private AppContextListener(final File rootFolder, final Module[] overrides) {
+            protected AppContextListener(final File rootFolder, final Module... overrides) {
                 this.rootFolder = rootFolder;
                 this.overrides = overrides;
             }

--- a/src/main/java/me/moodcat/database/DbModule.java
+++ b/src/main/java/me/moodcat/database/DbModule.java
@@ -89,7 +89,7 @@ public class DbModule extends AbstractModule {
          */
         private static final long serialVersionUID = 1071139882151857332L;
 
-        private DatabaseConfigurationException(final String string) {
+        protected DatabaseConfigurationException(final String string) {
             super(string);
         }
 

--- a/src/main/java/me/moodcat/database/controllers/AbstractDAO.java
+++ b/src/main/java/me/moodcat/database/controllers/AbstractDAO.java
@@ -21,7 +21,7 @@ public abstract class AbstractDAO<T> {
     /**
      * Manager that can talk to the actual database.
      */
-    private EntityManager entityManager;
+    private final EntityManager entityManager;
 
     protected AbstractDAO(final EntityManager entityManager) {
         this.entityManager = entityManager;

--- a/src/main/java/me/moodcat/database/controllers/SongDAO.java
+++ b/src/main/java/me/moodcat/database/controllers/SongDAO.java
@@ -44,26 +44,11 @@ public class SongDAO extends AbstractDAO<Song> {
                 .where(song.id.eq(id))
                 .singleResult(song);
     }
-    
+
     @Transactional
     public Song findBySoundCloudId(final int id) {
         return this.query().from(song)
                 .where(song.soundCloudId.eq(id))
                 .singleResult(song);
-    }
-
-    /**
-     * Fetches the next {@link #NUMBER_OF_SONGS_TO_PROCESS} in order to process to determine the
-     * {@link Song#getValenceArousal() valenceArousal}.
-     *
-     * @return A list consisting of songs that should be processed with a maximum size of
-     *         {@link #NUMBER_OF_SONGS_TO_PROCESS}
-     */
-    @Transactional
-    public List<Song> findNextUnprocessedSongs() {
-        return this.query().from(song)
-                .where(song.valenceArousal.isNull())
-                .limit(NUMBER_OF_SONGS_TO_PROCESS)
-                .list(song);
     }
 }

--- a/src/main/java/me/moodcat/database/embeddables/VAVector.java
+++ b/src/main/java/me/moodcat/database/embeddables/VAVector.java
@@ -109,9 +109,9 @@ public class VAVector {
     /**
      * Get the distance between two vectors.
      *
-     * @param a
+     * @param one
      *            first vector
-     * @param b
+     * @param other
      *            second vector
      * @return the distance
      */

--- a/src/main/java/me/moodcat/mood/Mood.java
+++ b/src/main/java/me/moodcat/mood/Mood.java
@@ -2,6 +2,7 @@ package me.moodcat.mood;
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
 import java.util.stream.Collectors;
 
 import lombok.Getter;
@@ -45,13 +46,13 @@ public enum Mood {
      */
     @Getter
     @JsonIgnore
-    public final VAVector vector;
+    private final VAVector vector;
 
     /**
      * Readable name for the frontend.
      */
     @Getter
-    public final String name;
+    private final String name;
 
     Mood(final VAVector vector, final String name) {
         this.vector = vector;
@@ -91,7 +92,7 @@ public enum Mood {
     public static VAVector createTargetVector(final List<String> moods) {
         final List<VAVector> actualMoods = moods.stream()
                 .filter(Mood::nameRepresentsMood)
-                .map(mood -> Mood.valueOf(mood.toUpperCase()))
+                .map(mood -> Mood.valueOf(mood.toUpperCase(Locale.ROOT)))
                 .map(mood -> mood.getVector())
                 .collect(Collectors.toList());
 

--- a/src/main/java/me/moodcat/soundcloud/SoundCloudAPIConnector.java
+++ b/src/main/java/me/moodcat/soundcloud/SoundCloudAPIConnector.java
@@ -5,7 +5,7 @@ import javax.ws.rs.client.Client;
 import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
 
 /**
- * Connects to the SoundCloud API using its {@link #urlFactory}.
+ * Connects to the SoundCloud API using its {@link #createClient()}.
  *
  * @author JeremyBellEU
  */

--- a/src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java
+++ b/src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java
@@ -98,9 +98,14 @@ public class SoundCloudExtract extends SoundCloudAPIConnector {
          * when you only know the SoundCloud.com URL.
          *
          * @param url
-         *            the url to retrieve
+         *            The url to retrieve.
+         * @param targetEntity
+         *            What entity type should be returned.
+         * @param <T>
+         *            The type of the entity.
+         * @return The entity that corresponds to the request.
          * @throws SoundCloudException
-         *             if the resource could not be accessed
+         *             Thrown if the resource could not be accessed.
          */
         protected <T> T resolve(final String url, final Class<T> targetEntity)
                 throws SoundCloudException {

--- a/src/main/resources/configuration/pmd-rules.xml
+++ b/src/main/resources/configuration/pmd-rules.xml
@@ -1,0 +1,26 @@
+<?xml version="1.0"?>
+
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	name="Moodcat PMD ruleset" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+	xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+	<description>
+        MoodCat PMD rules to apply on the codebase.
+    </description>
+	<rule ref="rulesets/java/basic.xml"/>
+    <rule ref="rulesets/java/braces.xml"/>
+    <rule ref="rulesets/java/codesize.xml"/>
+    <rule ref="rulesets/java/coupling.xml">
+    	<!-- Inheritly happening when using Guice, therefore these errors are not useful -->
+    	<exclude name="LawOfDemeter"/>
+    	<!-- Is a broken rule -->
+    	<exclude name="LoosePackageCoupling"/>
+    </rule>
+    <rule ref="rulesets/java/design.xml"/>
+    <rule ref="rulesets/java/empty.xml"/>
+    <rule ref="rulesets/java/imports.xml"/>
+    <rule ref="rulesets/java/strings.xml"/>
+    <rule ref="rulesets/java/unusedcode.xml">
+    	<!-- Lombok will cause errors for PMD, so we can safely exlude this one -->
+    	<exclude name="UnusedPrivateField"/>
+    </rule>
+</ruleset>

--- a/src/test/java/me/moodcat/core/AppTest.java
+++ b/src/test/java/me/moodcat/core/AppTest.java
@@ -2,12 +2,14 @@ package me.moodcat.core;
 
 import static org.junit.Assert.assertNotNull;
 
+import java.io.IOException;
+
 import org.junit.Test;
 
 public class AppTest {
 
     @Test
-    public void appCanBeLaunched() {
+    public void appCanBeLaunched() throws IOException {
         assertNotNull(new App());
     }
 }

--- a/src/test/java/me/moodcat/soundcloud/SoundCloudIntegrationTest.java
+++ b/src/test/java/me/moodcat/soundcloud/SoundCloudIntegrationTest.java
@@ -34,17 +34,17 @@ public class SoundCloudIntegrationTest {
 
     @BeforeClass
     public static void initializeTrack1() throws SoundCloudException {
-        SoundCloudExtract soundCloudExtract = new SoundCloudExtract();
+        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
+        final SoundCloudExtract soundCloudExtract = new SoundCloudExtract();
         testTrack = soundCloudExtract.extract(TRACK_URL);
     }
 
     @Test
     public void testSearch() throws SoundCloudException {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
-        SoundCloudSearch search = new SoundCloudSearch();
-        List<SoundCloudTrack> tracks = search.search("Katfyr");
+        final SoundCloudSearch search = new SoundCloudSearch();
+        final List<SoundCloudTrack> tracks = search.search("Katfyr");
         assertTrue(tracks.size() > 10);
-        for (SoundCloudTrack track : tracks) {
+        for (final SoundCloudTrack track : tracks) {
             assertTrue(track.getId() > 0);
             assertTrue(track.getPermalink() != null);
         }
@@ -52,43 +52,36 @@ public class SoundCloudIntegrationTest {
 
     @Test
     public void testExtractTitle() throws SoundCloudException {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getTitle(), TRACK_TITLE);
     }
 
     @Test
     public void testExtractPermalink() throws SoundCloudException {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getPermalink(), TRACK_PERMALINK);
     }
 
     @Test
     public void testExtractUser() throws SoundCloudException {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getUser().getUsername(), TRACK_USER);
     }
 
     @Test
     public void testExtractId() {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getId(), TRACK_ID);
     }
 
     @Test
     public void testArtworkUrl() {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getArtworkUrl(), TRACK_ARTWORK_URL);
     }
 
     @Test
     public void testDuration() {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.getDuration(), TRACK_DURATION);
     }
 
     @Test
     public void testDownloadable() {
-        Assume.assumeTrue("true".equals(System.getProperty("runWithIntegration")));
         assertEquals(testTrack.isDownloadable(), TRACK_DOWNLOADABLE);
     }
 }

